modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequest.java

```diff
@@ -58,4 +60,21 @@ public ClientRequest(long reqId) {
     public ClientResponse process(ClientConnectionContext ctx) {
         return new ClientResponse(reqId);
     }
+
+    /**
+     * Processes the request asynchronously.
+     *
+     * @return Future for response.
+     */
+    public IgniteInternalFuture<ClientResponse> processAsync(ClientConnectionContext ctx) {
+        return new GridFinishedFuture<>(process(ctx));
```

Comment:
This should not happen for synchronous requests - should we throw an exception?

Reply:
Fixed